src/match_data_object.cpp

```diff
@@ -2,6 +2,21 @@
 
 namespace Natalie {
 
+Value MatchDataObject::array(int start) {
+    auto size = (size_t)(m_region->num_regs - start);
```

Comment:
Should we throw a check in here (either an assertion or a raise) if the start is too big?